### ISSUE TITLE
clickjacking protection misconfiguration

### ISSUE THREAD

**Issue by ninabreznik** [playproject-io/playproject-io.github.io#13]
Just got this email

Hi team,

While performing security testing of your website i have found the vulnerability called Clickjacking.
Many URLS are in scope and vulnerable to Clickjacking.

What is Clickjacking ?
Clickjacking (User Interface redress attack, UI redress attack, UI redressing) is a malicious technique of tricking a Web user into clicking on something different from what the user perceives they are clicking on, thus potentially revealing confidential information or taking control of their computer while clicking on seemingly innocuous web pages.
The server didn't return an X-Frame-Options header which means that this website could be at risk of a clickjacking attack. The X-Frame-Options HTTP response header can be used to indicate whether or not a browser should be allowed to render a page in a <frame> or <iframe>. Sites can use this to avoid clickjacking attacks, by ensuring that their content is not embedded into other sites.
This vulnerability affects Web Server.

Steps to Reproduce / POC

Vulnerable Urls:https://playproject.io/
Put every above url one by one in the code of iframe, which is given below

```html
<!DOCTYPE html>
<html>
<body>

<h1>The iframe element</h1>

<iframe src="https://playproject.io/">
  <p>Your browser does not support iframes.</p>
</iframe>

</body>
</html>
```

By hasan khan
hasankhan222424@gmail. Com

Notice that site is visible in the Iframe

POC is in the attachments. Thanks, waiting for your response.

Impact

Using a similar technique, keystrokes can also be hijacked. With a carefully crafted combination of stylesheets, iframes, and text boxes, a user can be led to believe they are typing in the password to their email or bank account, but are instead typing into an invisible frame controlled by the attacker
![Screenshot_2020-05-18-11-31-59-452_com android chrome](https://user-images.githubusercontent.com/1076427/82235760-113bda00-992b-11ea-9f94-2c653753fd03.jpg)

**Comment by serapath** [comment]
lol :-)
joker.
We are allowing this on purpose.

**attack scenario**
So it all boils down to malicious developers who would embed our page into an iframe and then creating an overlay over it so our page is hidden under that and not visible.
The overlay could look more or less like our page so the user would start clicking and typing assuming he is interacting with our page, but instead they interact with the page of the malicious developer who embeds our page into an iframe, thus allowing them to read e.g. passwords or other things the user might type into input fields of the parent page while thinking they are typing into our page.

**useful scenario**
we are trying to generally allow anyone to embed our pages into iframes to interact with them, so disallowing that would defeat the purpose we are after in the first place.
It also doesn't matter because our page does not prompt users for anything sensitive.

**mitigation technique**
instead of disallowing a page to be embedded into an iframe, we would detect if a page is being embedded into an iframe and maybe de-activate such sensitive features

**probolem**
The general problem is, that a malicious developer can always proxy our page and re-write any head, so even if we'd follow the above technique, it would not protect and in the end it's up to the end user to check that the URL they are visiting is `https://playproject.io` and not something else, because in the above described scenario - anyway - the main domain would not be ours but a different one owned by the attacker.

---
**in summary**
won't fix, we can just close this. It's not relevant at all.
